[RTM] FIX: Correctly resample T1w images in ConformSeries

effigies opened the pull request:
This PR stems from #599, which revealed that we've been conforming wrongly. The error was so small that the error only appears for highly anisotropic voxels in an oblique orientation.

In addition, it introduced the case where high- and low-quality T1w scans coexist in the same dataset, and the upsampling involved would ultimately degrade the quality of a template produced by averaging the images.

The following changes are introduced:

* Rescaling is applied to affine columns, as opposed to rows, which will correctly set the zooms in the resampled dataset
* A size (dimension) factor is recalculated to be less sensitive to large differences in dimension
* A `max_scale` factor (default: 3) discards images which require upsampling over a given ratio
* An HTML reportlet is created to describe the conformation step

Todo:

* [x] Integrate reportlet into report
* [x] Test on ds000114 sub-02 (See #538/#545)
* [x] Test on ds000114 sub-07/09 (See #562/#564)
* [x] Test on ds000031

Closes #599.

effigies commented:
Reportlet of ds31, where 7 images were discarded:

![ds31_conformation](https://user-images.githubusercontent.com/83442/28627301-5c7b36a2-71ef-11e7-8629-cdf14ea332d4.png)

Displayed the basename, with the full (albeit within-docker) path as hover text.

effigies commented:
Integrated reports:

* https://654-78558509-gh.circle-artifacts.com/1/home/ubuntu/ds005/out/fmriprep/sub-01.html
* https://654-78558509-gh.circle-artifacts.com/0/home/ubuntu/ds054/out/fmriprep/sub-100185.html

chrisfilo commented:
The code and reports look great, but could you also add a description of this procedure to the documentation?

effigies commented:
Added [description](https://663-78558509-gh.circle-artifacts.com/0/home/ubuntu/docs/workflows.html#t1w-t2w-preprocessing) with link to [ConformSeries API entry](https://663-78558509-gh.circle-artifacts.com/0/home/ubuntu/docs/anat/base.html#fmriprep.interfaces.images.ConformSeries).

Updated [sample report](https://663-78558509-gh.circle-artifacts.com/0/home/ubuntu/docs/_static/sample_report.html).
